tests/test_graphql_fetch.py

```
@@ -0,0 +1,193 @@
+    page_one = _make_graphql_response(
+        [{"node": {"title": "Role 1", "applyUrl": "https://x/1", "postedAt": "2026-01-01", "description": "d1", "locations": [{"name": "Remote"}]}}],
+        has_next_page=True,
+        end_cursor="cursor-1",
+    )
+    page_two = _make_graphql_response(
+        [{"node": {"title": "Role 2", "applyUrl": "https://x/2", "postedAt": "2026-01-02", "description": "d2", "locations": [{"name": "Austin, TX"}]}}],
+        has_next_page=False,
+        end_cursor=None,
+    )
+
+    def _side_effect(*args, **kwargs):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_side_effect`

(ANN202)